homeassistant/components/harmony/remote.py

```diff
@@ -331,7 +282,7 @@ async def got_disconnected(self, _=None):
         self._available = False
         # We're going to wait for 10 seconds before announcing we're
         # unavailable, this to allow a reconnection to happen.
-        await asyncio.sleep(10)
+        await self.sleep(10)
```

Comment:
Can we use `asyncio.Event` here instead so its more reliable?

Reply:
Sorry, I'm struggling a bit to understand your suggestion. I'm a bit newer to async/await style concurrency, so sorry if I'm missing something. :smile: I tried grepping around the repo quite a bit and it seems to be pretty common to just await on chains of async functions and should reliably wait until the `await asyncio.sleep` call completes within our `sleep` wrapper.

Reply:
Let's do something like this instead and avoid the sleep:

```
TIME_MARK_DISCONNECTED = 10

```
```
self._unsub_mark_disconnected = async_call_later(self.hass, TIME_MARK_DISCONNECTED, self._mark_disconnected_if_unavailable)
```

Then call `self._unsub_mark_disconnected` if its not `None` when removing the entity.

Reply:
Makes a ton of sense - thank you! Should be updated.